### ISSUE TITLE
[Proposal] Add `--skip-goldens` to `flutter test` options

### ISSUE THREAD

**Issue by g0rdan** [flutter/flutter#156150]
### Use case

On many occasions, I need to run `flutter test` in repositories I'm unfamiliar with or new to them. Personally, >80% of the time these repositories have golden tests that are failing due to the fact that I run them on my local machine which is different from the machine they were generated even if there is a certain tolerance value. It produces a lot of failures that aren't relevant.

As far as I understand the issue could be solved locally, for example, by utilizing test's tags and `--dart-define` or, maybe, by writing your own `GoldenFileComparator` but all of it requires modification of existing code and, the proposal is to have out-of-box solution for it.

### Proposal

Add `--skip-goldens` option to skip entirely all golden tests in a directory/project.

**Comment by christopherfujino** [comment]
Although this sounds like a reasonable request, unfortunately I don't think the Flutter CLI tool will ever implement this. Unfortunately, there is no easy & robust way for the Flutter tool to know which of your tests are golden tests.

Even if we used package:analyzer to detect instances of the `GoldenFileComparator` class, I'm not sure how we would associate those references with a particular test name (we could detect the nearest surrounding closure, but that doesn't mean the string for the name of the test is near by). And we would need to determine the name of the test, and then tell `package:test` to run all tests *except* for those that referenced `GoldenFileComparator`. Also, I think running `package:analyzer` on every test library before actually invoking tests would significantly slow down the setup before running tests.

In summary, the tool does not know about individual tests, much less whether each test is a golden test (it operates on test *libraries*, and passes these libraries to `package:test`), so it's not really feasible to filter these out via a flag.

Again sorry, but thanks for contributing!

**Comment by g0rdan** [comment]
@christopherfujino Thanks for the response!
